lib/cc/analyzer/formatters/json_formatter.rb

```diff
@@ -3,18 +3,36 @@ module Analyzer
     module Formatters
       class JSONFormatter < Formatter
 
+        def initialize
+          @has_begun = false
+        end
+
         def engine_running(engine)
           @active_engine = engine
           yield
           @active_engine = nil
         end
 
+        def started
+          print "[ "
+        end
+
+        def finished
+          print " ]\n"
+        end
+
         def write(data)
           return unless data.present?
 
           document = JSON.parse(data)
           document["engine_name"] = @active_engine.name
-          puts document.to_json
+
+          if @has_begun
```

Comment:
@ABaldwinHunter sorry - was heads down in bugsnag hell. Whichever you think is best is cool with me!